src/globus_cli/utils.py

```diff
@@ -0,0 +1,34 @@
+from typing import Dict
+
+
+def format_list_of_words(word: str, *other: str):
+    if not other:
+        return word
+    if len(other) == 1:
+        return f"{word} and {other[0]}"
+    return ", ".join([word] + list(other[:-1])) + f", and {other[-1]}"
+
+
+def format_plural_str(formatstr: str, pluralizable: Dict[str, str], use_plural: bool):
+    """
+    Format text with singular or plural forms of words. Use the singular forms as
+    keys in the format string.
+
+    Usage:
+
+    >>> command_list = [...]
+    >>> fmtstr = "you need to run {this} {command}:"
+    >>> print(
+    ...     format_plural_str(
+    ...         fmtstr,
+    ...         {"this": "these", "command": "commands"},
+    ...         len(command_list) == 1
+    ...     )
+    ... )
+    >>> print("  " + "\n  ".join(command_list))
+    """
+    argdict = {
+        singular: plural if use_plural else singular
+        for singular, plural in pluralizable.items()
+    }
+    return formatstr.format(**argdict)
```

Comment:
Could "onelinerize" `argdict` into here.

Reply:
Yeah, I considered... But I think this is easier to read? Opinions on this probably vary.